encode_value.go

```diff
@@ -156,7 +156,7 @@ func encodeInterfaceValue(e *Encoder, v reflect.Value) error {
 }
 
 func encodeErrorValue(e *Encoder, v reflect.Value) error {
-	if v.IsNil() {
+	if v.IsNil() || !v.CanInterface() {
```

Comment:
This should be a separate check like

```
	if !v.CanInterface() {
		return fmt.Errorf("msgpack: Encode(non-interfaceable %s)", v.Type())
	}
```

Probably this is all we can do, because it looks like you can't use `Interface` on fields of private embedded struct. E.g. if you rename `interfaceAlias` to `InterfaceAlias` your example works fine.

Reply:
I'd really prefer for this not to be an error. `encoding/json` silently ignores this case, just like with an unexported scalar field. https://play.golang.org/p/OsKnVaNrnsL

Although: maybe it would be better to just do nothing rather than `EncodeNil()`.

It's relevant to my situation because I'm not in full control of the data people are passing in, and the expectation is that anything `encoding/json` can handle, this can handle as well. Every time we have a case where that isn't true, it creates a new surprise/bug/change requirement. I ran into this with a case where the pre-existing expectation is that the unexported embedded error is ignored.